bundles/core/src/main/java/com/adobe/cq/commerce/core/search/internal/models/SearchOptionsImpl.java

```diff
@@ -0,0 +1,121 @@
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.core.search.internal.models;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.adobe.cq.commerce.core.search.models.SearchOptions;
+
+public class SearchOptionsImpl implements SearchOptions {
+
+    private static final Integer PAGE_SIZE_DEFAULT = 6;
+    private static final String CATEGORY_ID_PAREMETER_ID = "category_id";
+    private static final String SEARCH_QUERY_PAREMETER_ID = "search_query";
+
+    Map<String, String> attributeFilters;
+
+    String categoryId;
+
+    String searchQuery;
+
+    Integer currentPage = 1;
+
+    Integer pageSize = PAGE_SIZE_DEFAULT;
+
+    public SearchOptionsImpl() {
+        attributeFilters = new HashMap<>();
+    }
+
+    @Override
+    public Map<String, String> getAllFilters() {
+        Map<String, String> allFilters = new HashMap<>(getAttributeFilters());
+
+        if (getCategoryId().isPresent()) {
+            allFilters.put(CATEGORY_ID_PAREMETER_ID, getCategoryId().get());
+        }
+        if (getSearchQuery().isPresent()) {
+            allFilters.put(SEARCH_QUERY_PAREMETER_ID, getSearchQuery().get());
+        }
+
+        return allFilters;
+    }
+
+    @Override
+    public Map<String, String> getAttributeFilters() {
+        return attributeFilters;
+    }
+
+    public void setAttributeFilters(final Map<String, String> attributeFilters) {
+        this.attributeFilters = attributeFilters;
+    }
+
+    @Override
+    public Optional<String> getCategoryId() {
+        return Optional.ofNullable(categoryId);
+    }
+
+    public void setCategoryId(final String categoryId) {
+        this.categoryId = categoryId;
+    }
+
+    @Override
+    public Optional<String> getSearchQuery() {
+        return Optional.ofNullable(searchQuery);
+    }
+
+    public void setSearchQuery(final String searchQuery) {
+        this.searchQuery = processSearchTerm(searchQuery);
+    }
+
+    @Override
+    public int getCurrentPage() {
+        return currentPage;
+    }
+
+    public void setCurrentPage(final Integer currentPage) {
+        this.currentPage = currentPage;
+    }
+
+    @Override
+    public int getPageSize() {
+        return pageSize;
+    }
+
+    public void setPageSize(final Integer pageSize) {
+        this.pageSize = pageSize;
+    }
+
+    /**
+     * Processes the search term to prepare it for the actual query.
+     * This method just prepends/appends the default wildcard character % to the search term. Overriding methods can implement their own
+     * processing.
+     *
+     * @return the processed search term, by default {@code "%searchTerm%"}
+     */
+    @Nonnull
+    private String processSearchTerm(String searchTerm) {
```

Comment:
This seems to be obsolete, you no longer add the % characters, right? Can we remove that?

Reply:
Originally this was done as it gave "wildcard" matches back to the search results, but you're right, the comment at least is obsolete because from what I can tell from the current Magento documentation:

> Wildcards are no longer supported in products queries.

I can / will remove this method, but I guess we should just be aware that in that case there is no filtering done on the search query (which I suppose is probably best, allow Magento to reject / deal with invalid query terms)